### PR TITLE
feat: support include other nginx config

### LINKED ISSUE #1620: [DISCUSS] "config.yaml" supports annotation configuration
### Issue description

The current nginx.conf configuration file for apache apisix is generated by `bin/apisix` from the `config.yaml` configuration file, so nginx.conf is best generated by the `bin/apisix` script rather than maintaining a copy itself.

However, the current `bin/apisix` support for generating `nginx.conf` is a generic configuration and does not support some custom or temporary configurations.

It is hoped that `config.yaml` will add an annotation field and that the configuration of this part will be generated directly to the specified location in `nginx.conf`.

Similar practice: in kubernetes-ingress, Ingress configuration has a comment field, where you can fill in large `nginx.conf` custom configurations directly in the comment field, instead of having to implement a piece of code for each field (because some configurations are customized, and it doesn't make sense to leave them to `bin/apisix` implementation, which is not general enough).

### Environment

* apisix version (cmd: `apisix version`):
* OS:

### LINKED ISSUE #2565: support include other nginx config in the apisix(lua file)
### Issue description
I can find that apisix use template to generate nginx.conf in the apisix(lua file),   we need to change this file  in following scenarios in my company
- Setting third nginx module param;
- Setting nginx params not supported in config.ymal
- Export other servers  and locations which third nginx module needed

What's about adding a new feature, for example,  support including other nginx conf file in the  apisix(lua file)
Thank you!

### Environment

* apisix version (master):

### PR DESCRIPTION
**Pull request by spacewander**
Close #2565
Close #1620

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible?

### CODE REVIEW

File: .travis/apisix_cli_test.sh [hunk 1]
```diff
@@ -458,6 +458,64 @@ fi
...
+
+echo '
+apisix:
+    node_listen: 9080
+    enable_admin: true
+    port_admin: 9180
+    stream_proxy:
+        tcp:
+            - 9100
+nginx_config:
```
Comment (membphis) [inline .travis/apisix_cli_test.sh:471]: How about this style? I think one level is enough and it looks simpler.

```yaml
nginx_config:
    main_configuration_snippet: |
        daemon on;
    http_configuration_snippet: |
        chunked_transfer_encoding on;
    http_server_configuration_snippet: |
        set $my "var";
    http_admin_configuration_snippet: |
        log_format admin "$request_time $pipe";
    stream_configuration_snippet: |
        tcp_nodelay off;

Comment (spacewander) [reply .travis/apisix_cli_test.sh:471]: @membphis 
Changed.

**Review by membphis — APPROVED** [review]

**Comment by nic-chen** [comment]
should add some doc to guid users.

**Comment by spacewander** [comment]
@nic-chen @membphis 
Doc added.

**Comment by membphis** [comment]
> @nic-chen @membphis
> Doc added.

Add a reference to https://github.com/apache/apisix/blob/master/doc/README.md .

**Comment by spacewander** [comment]
@membphis 
Solved.

**Comment by membphis** [comment]
@nic-chen please take a look when you have time

**Review by nic-chen — APPROVED** [review]

**Pull request merged**